File: spark/src/main/scala/com/jason/spark/core/rdd/Spark14_Action_Frist.scala
package com.jason.spark.core

import org.apache.spark.{SparkConf, SparkContext}

object Spark14_Action_Frist {
  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark11_KeyValue_Operation")
    val sc = new SparkContext(sparkConf)
    val dataRDD = sc.makeRDD(List(1, 2, 3, 4))
    val i = dataRDD.first()
    println(i)
  }

}

File: spark/src/main/java/com/jason/spark/core/rddtest/Spark14ActionFrist.java
package com.jason.spark.core.rddtest;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;

/**
 * @Description:
 * @author: 贾森
 * @date: 2024年10月28日 15:11
 */
public class Spark14ActionFrist {
    public static void main(String[] args) {
        // 1.创建配置对象
        SparkConf conf = new SparkConf().setMaster("local[*]").setAppName("sparkCore");

        // 2. 创建sparkContext
        JavaSparkContext sc = new JavaSparkContext(conf);

        // 3. 编写代码
        JavaRDD<Integer> integerJavaRDD = sc.parallelize(Arrays.asList(1, 2, 3, 4),2);
        Integer first = integerJavaRDD.first();
        System.out.println(first);
    }
}
